### PR TITLE
ni_beds

### PR DESCRIPTION
**Pull request by niuni02**
Found a dataset showing bed availability and occupancy in Northern Ireland from a health view (Health and Social Care Trusts). Calculated discharged beds using bed availability and occupancy data. Added the roxygen comments into R/data.R and then built documentation for the dataset in man/.

### CODE REVIEW

**Comment by MikeJohnPage** [comment]
This data set shares a lot of the same indicators as the `ni_beds_occupancy` dataset, albeit with more information. Are there any particular indicators you were hoping to use in the HIE? What was the motivation for adding this second bed data set?

It might be worth keeping this data set instead of the other one, given it has a more detailed breakdown of beds over time, but there are quite a few changes we would need to make. Let me know your thoughts on the above questions and then we can come to a decision.

**Comment by niuni02** [comment]
I managed to find this dataset and thought it was relevant since it seemed more up to date as it included multiple indicators from the web app (bed availability, bed occupancy and discharged beds). Also, since I could not find a separate dataset with ‘discharged beds’ information, I thought it would be relevant to calculate the ‘total discharged beds’ using the formula: discharged beds = total beds available - total beds occupied. Furthermore, I thought it would be more efficient to include all the ‘hospital bed’ information in one dataset so that in the long-term it would be less messy. I was also trying to replicate the scotland_beds dataset for consistency. But I understand if it is preferred to separate the indicators and I can do this if you would like me to.

In terms of the indicators, I thought that the ‘total_inpatients’ data was interesting since if we can find another dataset that has data on outpatients, a comparison could be make to determine the efficiency of allocated resources. Also the distinction between ‘elective_inpatients’ and ‘non_elective_inpatients’ could be intriguing since we could investigate the pressures of emergencies in different HSCs - although this could easily overcomplicate the app so may not be as relevant.

I am open to keeping this dataset if you prefer and making any changes you require. Please let me know what you think!

File: data-raw/ni-beds.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  query_urls |>
+  filter(id == "ni_beds") |>
+  pull(query)
+
+# ---- Download and wrangle data ----
+raw <- read_csv(query_url, skip=2)
+
+# Change dataset to tidy data with all relevant information
+# Removed "Day Case" column since it included missing values and duplicate values with "Total Day Cases".
+# Can calculate number of discharge beds = total available beds - total occupied beds
```
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:22]: Where possible, we do not want to perform our own calculations in the humaniverse data packages, and try and ship the data as close to its raw form as possible. Calculations can be added in the projects where they are used.

Comment (niuni02) [reply data-raw/ni-beds.R:22]: Yes, that makes sense. Would you like me to remove the discharged beds information and then try to find a separate dataset for that information?

Comment (MikeJohnPage) [reply data-raw/ni-beds.R:22]: I'm just going to finish the code review now with some more comments 👍

File: data-raw/ni-beds.R
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:1]: Please can we make sure all files are styled with the [`styler`](https://github.com/r-lib/styler) R package into a [tidyverse style](https://style.tidyverse.org/) before submitting for review

File: data-raw/ni-beds.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ni_beds <- raw |>
+  select(
+    financial_year = "Financial Year",
```
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:25]: The financial year col isn't needed, as the quarter ending column capture this information.

File: data-raw/ni-beds.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    quarter_ending = "Quarter Ending",
```
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:26]: Please can we encode this variable as class `date` and not `character`. The function `lubridate::as_date()` might come in handy here.

File: data-raw/ni-beds.R [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    HSC = "HSC Trust",
```
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:27]: Please match this column name to the same key used throughout the other humaniverse packages, e.g., use `trust18_code` instead.

File: data-raw/ni-beds.R (hunk 1 above)
Comment (MikeJohnPage) [reply data-raw/ni-beds.R:22]: > Yes, that makes sense. Would you like me to remove the discharged beds information and then try to find a separate dataset for that information?

Yes, please remove this compute variable, and add a separate data set if it exists. However, I don't this data is available after our previous team discussions last week.

File: R/data.R
Comment (MikeJohnPage) [inline R/data.R:1]: Don't forget to update and rebuild the documentation after the changes have been made.

**Review by MikeJohnPage — CHANGES_REQUESTED** [review]

File: data-raw/ni-beds.R (hunk 3 above)
Comment (MikeJohnPage) [reply data-raw/ni-beds.R:26]: You need to pass in a format:

```r
ni_beds |> 
  mutate(quarter_ending = as_date(quarter_ending, format = "%d/%m/%Y"))
```

File: data-raw/ni-beds.R [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  pull(query)
+
+# ---- Download and wrangle data ----
+raw <- read_csv(query_url, skip = 2)
+
+# Removed "Day Case" column since it included missing values and duplicate values with "Total Day Cases".
+ni_beds <- raw |>
+  select(
+    quarter_ending = "Quarter Ending",
+    trust_code = "HSC Trust",
```
Comment (MikeJohnPage) [inline data-raw/ni-beds.R:23]: `"HSC Trust"` isn't a code, it is a name. The trust codes you need can be found in `geographr::boundaries_trusts_ni18`:

Screenshot 2023-07-11 at 10 53 07

To add these codes to this data set you will need to create a lookup like so:

```r
library(sf)

lookup_trusts <- geographr::boundaries_trusts_ni18 |> 
  st_drop_geometry()

```
and then join this lookup to the tibble you have created. Please make the `trust18_code` the first column in the data set. Please drop the `trust18_name` column, as just the trust code is needed as a primary key to join to other data sets in the humaniverse and HIE.

File: data-raw/ni-beds.R (hunk 3 above)
Comment (MikeJohnPage) [reply data-raw/ni-beds.R:22]: Please can we rename `quarter_ending` to `date`, which is a more common name used throughout the humaniverse?

**Review by MikeJohnPage — CHANGES_REQUESTED** [review]

File: data-raw/ni-beds.R (hunk 5 above)
Comment (niuni02) [reply data-raw/ni-beds.R]: I have made these changes! Please let me know if there is anything else I need to do :)

Comment (MikeJohnPage) [reply data-raw/ni-beds.R]: Don't forget to hit "Revolve conversation" when you think your changes are ready! I will do it now.

**Comment by MikeJohnPage** [comment]
Merge conflicts resolved in a new branch. Closing this branch.

**Pull request closed without merging**